packages/block-library/src/table-of-contents/hooks.js

```diff
@@ -144,13 +145,24 @@ function observeCallback( select, dispatch, clientId ) {
 	}
 }
 
-export function useObserveHeadings( clientId ) {
+export function useObserveHeadings( clientId, headings ) {
 	const registry = useRegistry();
+	const blockType = useSelect(
+		( select ) =>
+			select( blocksStore ).getBlockType( 'core/table-of-contents' ),
+		[]
+	);
+	const isExample =
+		JSON.stringify( blockType.example.attributes.headings ) ===
+		JSON.stringify( headings );
```

Comment:
I struggled to find a better option to avoid the `useObserveHeadings` hook overriding arbitrary block example attributes. More than happy to hear better approaches if you have them 🙏